languages/js/src/Host.ts

```diff
@@ -159,13 +159,18 @@ export class Host {
         existing,
       });
     }
+
+    // TODO(gw) maybe we only want to support plain objects?
+    let fields = types || {};
+    if (!(fields instanceof Map)) fields = new Map(Object.entries(fields));
+
     const userType = new UserType({
       name: clsName,
       cls: cls,
       buildQuery: buildQuery || this.buildQuery,
       execQuery: execQuery || this.execQuery,
       combineQuery: combineQuery || this.combineQuery,
-      fields: types || new Map(),
+      fields: fields,
```

Comment:
I think this works in JS? Been bouncing between all 6 langs this week so my head's a bit spinny

```suggestion
      fields,
```

if so could do it for `cls: cls,` above, too

Reply:
Oooh let me check, I sure hope that works...

Reply:
👍 thanks!